schemaregistry/schemaregistry.yaml

```diff
@@ -0,0 +1,313 @@
+        '404':
+          description: Group not found
+    delete:
+      tags:
+      - 'groups'
+      summary: Deletes all schemas in group
+      description: Deletes all schemas under specified group name.
+      operationId: deleteSchemasByGroup
+      responses:
+        '204':
+          description: OK no content
+        '404':
+          description: Group not found
+  /schemagroups/{group-name}/schemas/{schema-name}:
+    parameters:
+      - name: group-name
+        in: path
+        description: schema group
+        required: true
+        schema:
+          type: string
+      - name: schema-name
+        in: path
+        description: schema name
+        required: true
+        schema:
+          type: string
+    post: 
+      summary: Register schema
+      description: >
+        Register schema.
+        If schema of specified name does not exist in specified group, schema is created at version 1.
+        If schema of specified name exists already in specified group, schema is created at latest version + 1.
+        If schema with identical content already exists, existing schema's ID is returned. 
+      operationId: createSchema
+      requestBody:
+        description: schema content
+        required: true
+        content:
+          application/json;format=avro:
+            schema:
+              $ref: '#/components/schemas/SchemaBytePayload'
+      responses:
+        '200':
+          $ref: '#/components/responses/SchemaIdResponse'
+        '400': 
+          description: Invalid request
+      tags:
+        - 'runtime'
+    get:
+      summary: Get latest version of schema
+      description: Get latest version of schema. 
+      operationId: getLatestSchema
+      responses: 
+        '200':
+          $ref: '#/components/responses/SchemaBytePayloadResponse'
+      tags:
+        - 'schemas'
+    delete:
+      summary: Delete schema
+      operationId: deleteSchema
+      responses:
+        '204':
+          description: OK no content
+        '404':
+          description: Matching schema not found
+      tags:
+        - 'schemas'
+  /schemagroups/{group-name}/schemas/{schema-name}/versions:
+    parameters:
+      - name: group-name
+        in: path
+        description: schema group
+        required: true
+        schema:
+          type: string
+      - name: schema-name
+        in: path
+        description: schema name
+        required: true
+        schema:
+          type: string
+    get:
+      summary: Get list of versions
+      description: Get list of versions for specified schema
+      operationId: getSchemaVersions
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json;format=avro:
+              schema:
+                type: array
+                items: 
+                  type: integer
+      tags:
+        - 'schemas'
+  /schemagroups/{group-name}/schemas/{schema-name}/versions/{version-number}:
```

Comment:
There might be a case for metadata to live at the version level of a schema. For example, when the schema was created, who created it, etc. I wonder if we need separate endpoints for the metadata vs fetching the raw schema itself.

Reply:
I think those extra metadata items can be exposed through metadata annotations or through an OPTIONS call if they are not part of the schema itself.